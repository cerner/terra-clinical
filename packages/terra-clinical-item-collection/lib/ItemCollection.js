'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _resizeObserverPolyfill = require('resize-observer-polyfill');

var _resizeObserverPolyfill2 = _interopRequireDefault(_resizeObserverPolyfill);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _breakpoints = require('terra-responsive-element/lib/breakpoints');

var _breakpoints2 = _interopRequireDefault(_breakpoints);

require('terra-base/lib/baseStyles');

require('./ItemCollection.scss');

var _CreateListView = require('./CreateListView');

var _CreateListView2 = _interopRequireDefault(_CreateListView);

var _CreateTableView = require('./CreateTableView');

var _CreateTableView2 = _interopRequireDefault(_CreateTableView);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var propTypes = {
  /**
  * The breakpoint to switch from a table view to a list view. Breakpoint options are tiny, small, medium, large, or huge.
  */
  breakpoint: _propTypes2.default.oneOf(['tiny', 'small', 'medium', 'large', 'huge']),
  /**
   * The styles to spread to the table. Table style options are isPadded and isStriped.
   **/
  tableStyles: _propTypes2.default.shape({
    isPadded: _propTypes2.default.bool,
    isStriped: _propTypes2.default.bool
  }),
  /**
   * The styles to spread to the list. List style options are isDivided and hasChevrons.
   **/
  listStyles: _propTypes2.default.shape({
    isDivided: _propTypes2.default.bool,
    hasChevrons: _propTypes2.default.bool
  }),
  /**
   * The array of hashes to be displayed as rows. Each hash can contain a startAccessory, endAccessory,
   * comment, array of displays and a itemStyles hash. The item style options are layout, textEmphasis,
   * isTruncated and accessoryAlignment.
   **/
  rows: _propTypes2.default.arrayOf(_propTypes2.default.shape({
    startAccessory: _propTypes2.default.element,
    displays: _propTypes2.default.arrayOf(_propTypes2.default.element),
    comment: _propTypes2.default.element,
    endAccessory: _propTypes2.default.element,
    itemStyles: _propTypes2.default.shape({
      layout: _propTypes2.default.oneOf(['oneColumn', 'twoColumns']),
      textEmphasis: _propTypes2.default.oneOf(['default', 'start']),
      isTruncated: _propTypes2.default.bool,
      accessoryAlignment: _propTypes2.default.oneOf(['alignTop', 'alignCenter'])
    })
  })),
  /**
   * A callback event that will be triggered when selection state changes.
   */
  onChange: _propTypes2.default.func
};

var defaultProps = {
  listStyles: undefined,
  tableStyles: undefined,
  rows: [],
  breakpoint: 'small',
  onChange: undefined
};

var ItemCollection = function (_React$Component) {
  _inherits(ItemCollection, _React$Component);

  function ItemCollection(props) {
    _classCallCheck(this, ItemCollection);

    var _this = _possibleConstructorReturn(this, (ItemCollection.__proto__ || Object.getPrototypeOf(ItemCollection)).call(this, props));

    _this.state = { display: 'table', selectedIndex: -1 };
    _this.setContainer = _this.setContainer.bind(_this);
    _this.handleResize = _this.handleResize.bind(_this);
    _this.handleSelection = _this.handleSelection.bind(_this);
    return _this;
  }

  _createClass(ItemCollection, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      var _this2 = this;

      this.resizeObserver = new _resizeObserverPolyfill2.default(function (entries) {
        _this2.handleResize(entries[0].contentRect.width);
      });
      this.resizeObserver.observe(this.container);
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      this.resizeObserver.disconnect(this.container);
      this.container = null;
    }
  }, {
    key: 'setContainer',
    value: function setContainer(node) {
      if (node === null) {
        return;
      } // Ref callbacks happen on mount and unmount, element will be null on unmount
      this.container = node.parentNode;
    }
  }, {
    key: 'handleSelection',
    value: function handleSelection(event, selectedIndex) {
      this.setState({ selectedIndex: selectedIndex });
      if (this.props.onChange) {
        this.props.onChange(event, selectedIndex);
      }
    }
  }, {
    key: 'handleResize',
    value: function handleResize(width) {
      var display = void 0;
      var breakpoints = (0, _breakpoints2.default)();
      if (width < breakpoints[this.props.breakpoint]) {
        display = 'list';
      } else {
        display = 'table';
      }

      if (this.state.display !== display) {
        this.setState({ display: display });
      }
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          listStyles = _props.listStyles,
          tableStyles = _props.tableStyles,
          rows = _props.rows,
          breakpoint = _props.breakpoint,
          customProps = _objectWithoutProperties(_props, ['listStyles', 'tableStyles', 'rows', 'breakpoint']);

      if (!rows || !rows.length) {
        return null;
      }

      var attributes = _extends({}, customProps);
      attributes.className = (0, _classnames2.default)(['terraClinical-ItemCollection', 'terraClinical-ItemCollection--' + this.state.display + 'View', attributes.className]);

      var collectionDisplay = void 0;
      if (this.state.display === 'table') {
        collectionDisplay = (0, _CreateTableView2.default)(rows, tableStyles, this.state.selectedIndex, this.handleSelection);
      } else if (this.state.display === 'list') {
        collectionDisplay = (0, _CreateListView2.default)(rows, listStyles, this.state.selectedIndex, this.handleSelection);
      }

      return _react2.default.createElement(
        'div',
        _extends({ ref: this.setContainer }, attributes),
        _react2.default.createElement(
          'h2',
          null,
          'Item Collection - ',
          this.state.display
        ),
        _react2.default.createElement(
          'h2',
          null,
          ' Item Selection is: ',
          this.state.selectedIndex,
          ' '
        ),
        collectionDisplay
      );
    }
  }]);

  return ItemCollection;
}(_react2.default.Component);

ItemCollection.propTypes = propTypes;
ItemCollection.defaultProps = defaultProps;

exports.default = ItemCollection;