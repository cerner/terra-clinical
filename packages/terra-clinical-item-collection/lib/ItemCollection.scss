@import '~terra-mixins';

.terraClinical-ItemCollection-column--accessory {
  width: 40px;
}
