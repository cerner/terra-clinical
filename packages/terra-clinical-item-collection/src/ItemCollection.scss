:local {
  .column-accessory,
  .list-accessory {
    width: 40px;
  }

  .content-accessory {
    vertical-align: middle !important;
  }
}
