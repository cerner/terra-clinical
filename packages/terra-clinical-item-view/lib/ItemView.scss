@import '~terra-mixins';
@import './variables';
@import './mixins';

// Temporary until it is added to application.
.terraClinical-ItemView {
  word-break: break-word;
  word-wrap: break-word;
}

.terraClinical-ItemView {
  display: flex;
  flex-flow: row nowrap;
  justify-content: flex-start;
  width: 100%;
}

// Accessory

.terraClinical-ItemView-accessory {
  // Accessory is of fixed size and does not shrink or grow.
  flex: 0 0 auto;
  margin: 5px;
  max-height: 60px;
  max-width: 60px;
  min-width: 0;
  overflow: hidden;
}

.terraClinical-ItemView-accessory--alignTop {
  align-self: flex-start;
}

.terraClinical-ItemView-accessory--alignCenter {
  align-self: center;
}

// Content Layout

.terraClinical-ItemView-body {
  flex: 1 1 auto;
  margin: 5px;
  min-width: 0;
  overflow: hidden; // VERY IMPORTANT FOR IE10
}

.terraClinical-ItemView-row {
  display: flex;
  width: 100%;
}

.terraClinical-ItemView-content {
  align-items: center;
  display: flex;
  overflow: hidden; // VERY IMPORTANT FOR IE10
}

.terraClinical-ItemView-text {
  flex: 1 1 auto;
  min-width: 10px; // Need min-width, as IE10 will completely squish display
}

.terraClinical-ItemView-text--inlineIcon {
  @include terra-padding-end(5px);
  flex: 0 0 auto;
  height: 20px;
  max-width: 20px;
  overflow: hidden;
}

.terraClinical-ItemView-text--inlineIcon > * {
  font-size: 1rem;
}

// Truncation

.terraClinical-ItemView--isTruncated .terraClinical-ItemView-text,
.terraClinical-ItemView-text--isTruncated {
  @include terra-clinical-text-truncate;
}

// Layouts

.terraClinical-ItemView--oneColumn {
  .terraClinical-ItemView-rowContainer {
    .terraClinical-ItemView-row {
      flex-flow: column nowrap;
      justify-content: flex-start;
    }

    .terraClinical-ItemView-content {
      width: 100%;
    }
  }
}

.terraClinical-ItemView--twoColumns {
  .terraClinical-ItemView-content:nth-child(odd) {
    flex: 1 1 auto;
    text-align: left;
  }

  .terraClinical-ItemView-content:nth-child(even) {
    @include terra-padding-start(5px);

    flex: 1 0 auto;
    justify-content: flex-end;

    // Set a max-width and disable flexbox-squishing on the right-most displays. This will ensure that they aren't
    // prematurely squished and maximize the available space for all labels.
    max-width: 60%;

    .terraClinical-ItemView-text {
      flex: 0 1 auto;
      min-width: 10px; // Need min-width, as IE10 will completely squish display
    }
  }
}

// We need to style content that appears out of the rowContainer (i.e. the comment).
.terraClinical-ItemView-content {
  .terraClinical-ItemView-text,
  .terraClinical-ItemView-text--inlineIcon {
    @include terra-clinical-text-body;
    color: $terra-clinical-item-view-content-secondary-color;
  }
}

//Emphasis Content Styles
.terraClinical-ItemView-content--primaryColor {
  .terraClinical-ItemView-text,
  .terraClinical-ItemView-text--inlineIcon {
    color: $terra-clinical-item-view-content-primary-color;
  }
}

.terraClinical-ItemView-content--primarySize {
  .terraClinical-ItemView-text,
  .terraClinical-ItemView-text--inlineIcon {
    @include terra-clinical-tiny-heading;
  }
}

.terraClinical-ItemView-content--secondaryColor {
  .terraClinical-ItemView-text,
  .terraClinical-ItemView-text--inlineIcon {
    color: $terra-clinical-item-view-content-secondary-color;
  }
}

.terraClinical-ItemView-content--secondarySize {
  .terraClinical-ItemView-text,
  .terraClinical-ItemView-text--inlineIcon {
    @include terra-clinical-text-body;
  }
}

// Text Styles

.terraClinical-ItemView-text--secondary {
  color: $terra-clinical-item-view-text-secondary-color !important;
}

.terraClinical-ItemView-text--attention {
  color: $terra-clinical-item-view-text-attention-color !important;
  font-weight: $terra-clinical-item-view-text-attention-weight !important;
}

.terraClinical-ItemView-text--strikeThrough {
  text-decoration: line-through !important;
}
