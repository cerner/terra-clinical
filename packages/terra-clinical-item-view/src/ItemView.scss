@import './mixins';

// Temporary until it is added to application.
:local {
  .item-view {
    display: flex;
    flex-flow: row nowrap;
    justify-content: flex-start;
    width: 100%;
    word-break: break-word;
    word-wrap: break-word;
  }

  // Accessory is of fixed size and does not shrink or grow.
  .accessory {
    flex: 0 0 auto;
    margin: 5px;
    max-height: 60px;
    max-width: 60px;
    min-width: 0;
    overflow: hidden;
  }

  .accessory > * {
    display: flex !important;
    max-height: 60px;
    max-width: 60px;
  }

  .accessory-align-top {
    align-self: flex-start;
  }

  .accessory-align-center {
    align-self: center;
  }

  // Content Structure
  .body {
    flex: 1 1 auto;
    margin: 5px;
    min-width: 0;
    overflow: hidden; // VERY IMPORTANT FOR IE10
  }

  .row {
    display: flex;
    width: 100%;
  }

  .content {
    align-items: flex-start;
    color: #64696c;
    display: flex;
    overflow: hidden; // VERY IMPORTANT FOR IE10
  }

  .is-truncated,
  .is-truncated [class*='ItemDisplay__text'] {
    @include terra-clinical-text-truncate;
  }

  // Layouts
  /* stylelint-disable selector-max-compound-selectors */
  .one-column {
    .row-container {
      display: flex;
      flex-flow: row wrap;

      .content {
        width: 100%;
      }
    }
  }
  /* stylelint-enable selector-max-compound-selectors */

  .two-columns {
    .content:nth-child(odd) {
      flex: 1 1 auto;
      justify-content: flex-start;
    }

    .content:nth-child(even) {
      flex: 1 0 auto;
      justify-content: flex-end;
      // Set a max-width and disable flexbox-squishing on the right-most displays. This will ensure that they aren't
      // prematurely squished and maximize the available space for all labels.
      max-width: 60%;
      padding-left: 5px;
    }
  }

  //Emphasis Content Styles
  .content-primary-color {
    color: #1c1f21;
  }

  .content-primary-size {
    @include terra-clinical-tiny-heading;
  }

  .content-secondary-color {
    color: #64696c;
  }

  .content-secondary-size {
    @include terra-clinical-text-body;
  }
}
