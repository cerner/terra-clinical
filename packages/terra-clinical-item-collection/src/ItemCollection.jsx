import React from 'react';
import PropTypes from 'prop-types';
import ResponsiveElement from 'terra-responsive-element';
import 'terra-base/lib/baseStyles';
import './ItemCollection.scss';

import createListView from './_CreateListView';
import createTableView from './_CreateTableView';

const propTypes = {
  /**
  * The breakpoint to switch from a table view to a list view. Breakpoint options are tiny, small, medium, large, or huge.
  */
  breakpoint: PropTypes.oneOf(['tiny', 'small', 'medium', 'large', 'huge']),
  /**
   * The styles to spread to the table. Table style options are isPadded and isStriped.
   */
  tableStyles: PropTypes.shape({
    isPadded: PropTypes.bool,
    isStriped: PropTypes.bool,
  }),
  /**
   * The styles to spread to the list. List style options are isDivided and hasChevrons.
   */
  listStyles: PropTypes.shape({
    isDivided: PropTypes.bool,
    hasChevrons: PropTypes.bool,
  }),
  /**
   * The array of hashes to be displayed as rows. Each hash can contain a startAccessory, endAccessory,
   * comment, array of displays, isSelected, and a itemStyles hash. The item style options are layout,
   * textEmphasis, isTruncated and accessoryAlignment.
   */
  rows: PropTypes.arrayOf(PropTypes.shape({
    startAccessory: PropTypes.element,
    displays: PropTypes.arrayOf(PropTypes.element),
    comment: PropTypes.element,
    endAccessory: PropTypes.element,
    isSelected: PropTypes.bool,
    itemStyles: PropTypes.shape({
      layout: PropTypes.oneOf(['oneColumn', 'twoColumns']),
      textEmphasis: PropTypes.oneOf(['default', 'start']),
      isTruncated: PropTypes.bool,
      accessoryAlignment: PropTypes.oneOf(['alignTop', 'alignCenter']),
    }),
  })),
  /**
   * A callback event that will be triggered when selection state changes.
   */
  onChange: PropTypes.func,
};

const defaultProps = {
  listStyles: undefined,
  tableStyles: undefined,
  rows: [],
  breakpoint: 'small',
  onChange: undefined,
};

class ItemCollection extends React.Component {
  static selectedIndexFromRows(rows) {
    if (!rows || !rows.length) {
      return -1;
    }
    for (let i = 0; i < rows.length; i += 1) {
      if (rows[i].isSelected) {
        return i;
      }
    }
    return -1;
  }

  constructor(props) {
    super(props);
    this.state = { display: 'table', selectedIndex: ItemCollection.selectedIndexFromRows(this.props.rows) };
    this.handleSelection = this.handleSelection.bind(this);
  }

  handleSelection(event, selectedIndex) {
    this.setState({ selectedIndex });
    if (this.props.onChange) {
      this.props.onChange(event, selectedIndex);
    }
  }

  render() {
    const { listStyles, tableStyles, rows, breakpoint, ...customProps } = this.props;

    if (!rows || !rows.length) {
      return null;
    }

    const tableView = createTableView(rows, tableStyles, this.state.selectedIndex, this.handleSelection);
    const tableDisplay = (
      <div {...customProps} data-terra-clinical-item-collection-table-view className={customProps.className}>
        {tableView}
      </div>
    );

    const listView = createListView(rows, listStyles, this.state.selectedIndex, this.handleSelection);
    const listDisplay = (
      <div {...customProps} data-terra-clinical-item-collection-list-view className={customProps.className}>
        {listView}
      </div>
    );

    const breakpointDisplay = {};
    breakpointDisplay[breakpoint] = tableDisplay;

    return (
      <ResponsiveElement
        responsiveTo="parent"
        defaultElement={listDisplay}
        {...breakpointDisplay}
      />
    );
  }
}

ItemCollection.propTypes = propTypes;
ItemCollection.defaultProps = defaultProps;

export default ItemCollection;
