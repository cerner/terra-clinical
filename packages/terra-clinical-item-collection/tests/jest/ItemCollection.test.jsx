import React from 'react';
import ItemView from 'terra-clinical-item-view';
import ItemCollection from '../../src/ItemCollection';

const item = {
  startAccessory: <p>Start Accessory</p>,
  displays: [<ItemView.Display text="Display 1" />, <ItemView.Display text="Display 2" />],
  comment: <ItemView.Comment text="test comment" />,
  endAccessory: <p>End Accessory</p>,
};

// Snapshot Tests
it('should render a default component', () => {
  const itemCollection = shallow(<ItemCollection />);
  expect(itemCollection).toMatchSnapshot();
});

it('should render with one row', () => {
  const itemCollection = shallow(<ItemCollection rows={[item]} />);
  expect(itemCollection).toMatchSnapshot();
});

it('should render with multiple rows', () => {
  const itemCollection = shallow(<ItemCollection rows={[item, item, item]} />);
  expect(itemCollection).toMatchSnapshot();
});

it('should render with a given breakpoint', () => {
  const itemCollection = shallow(<ItemCollection breakpoint="medium" rows={[item, item, item]} />);
  expect(itemCollection).toMatchSnapshot();
});
