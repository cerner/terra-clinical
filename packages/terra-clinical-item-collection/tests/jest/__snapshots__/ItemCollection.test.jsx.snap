// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a default component 1`] = `null`;

exports[`should render with a given breakpoint 1`] = `
<ResponsiveElement
  defaultElement={
    <div
      data-terra-clinical-item-collection-list-view={true}
    >
      <SingleSelectList
        hasChevrons={false}
        isDivided={false}
        onChange={[Function]}
      >
        <ListItem
          content={
            <ItemView
              accessoryAlignment="alignCenter"
              comment={
                <ItemComment
                  isTruncated={false}
                  text="test comment"
                />
              }
              displays={
                Array [
                  <ItemDisplay
                    isTruncated={false}
                    text="Display 1"
                  />,
                  <ItemDisplay
                    isTruncated={false}
                    text="Display 2"
                  />,
                ]
              }
              endAccessory={
                <p>
                  End Accessory
                </p>
              }
              isTruncated={false}
              layout="oneColumn"
              startAccessory={
                <p>
                  Start Accessory
                </p>
              }
              textEmphasis="default"
            />
          }
          isSelected={false}
        />
        <ListItem
          content={
            <ItemView
              accessoryAlignment="alignCenter"
              comment={
                <ItemComment
                  isTruncated={false}
                  text="test comment"
                />
              }
              displays={
                Array [
                  <ItemDisplay
                    isTruncated={false}
                    text="Display 1"
                  />,
                  <ItemDisplay
                    isTruncated={false}
                    text="Display 2"
                  />,
                ]
              }
              endAccessory={
                <p>
                  End Accessory
                </p>
              }
              isTruncated={false}
              layout="oneColumn"
              startAccessory={
                <p>
                  Start Accessory
                </p>
              }
              textEmphasis="default"
            />
          }
          isSelected={false}
        />
        <ListItem
          content={
            <ItemView
              accessoryAlignment="alignCenter"
              comment={
                <ItemComment
                  isTruncated={false}
                  text="test comment"
                />
              }
              displays={
                Array [
                  <ItemDisplay
                    isTruncated={false}
                    text="Display 1"
                  />,
                  <ItemDisplay
                    isTruncated={false}
                    text="Display 2"
                  />,
                ]
              }
              endAccessory={
                <p>
                  End Accessory
                </p>
              }
              isTruncated={false}
              layout="oneColumn"
              startAccessory={
                <p>
                  Start Accessory
                </p>
              }
              textEmphasis="default"
            />
          }
          isSelected={false}
        />
      </SingleSelectList>
    </div>
  }
  medium={
    <div
      data-terra-clinical-item-collection-table-view={true}
    >
      <Table
        isPadded={true}
        isStriped={true}
        style={
          Object {
            "tableLayout": "fixed",
          }
        }
      >
        <TableHeader
          style={
            Object {
              "visibility": "hidden",
            }
          }
        >
          <TableHeaderCell
            columnType="accessory"
          />
          <TableHeaderCell
            columnType="display"
          />
          <TableHeaderCell
            columnType="display"
          />
          <TableHeaderCell
            columnType="comment"
          />
          <TableHeaderCell
            columnType="accessory"
          />
        </TableHeader>
        <SingleSelectableRows
          onChange={[Function]}
        >
          <TableRow
            isSelectable={true}
            isSelected={false}
          >
            <TableCell
              className="content-accessory"
              content={
                <p>
                  Start Accessory
                </p>
              }
            />
            <TableCell
              content={
                <ItemDisplay
                  isTruncated={false}
                  text="Display 1"
                />
              }
              data-terra-clinical-item-collection-content="display"
            />
            <TableCell
              content={
                <ItemDisplay
                  isTruncated={false}
                  text="Display 2"
                />
              }
              data-terra-clinical-item-collection-content="display"
            />
            <TableCell
              content={
                <ItemComment
                  isTruncated={false}
                  text="test comment"
                />
              }
              data-terra-clinical-item-collection-content="comment"
            />
            <TableCell
              className="content-accessory"
              content={
                <p>
                  End Accessory
                </p>
              }
            />
          </TableRow>
          <TableRow
            isSelectable={true}
            isSelected={false}
          >
            <TableCell
              className="content-accessory"
              content={
                <p>
                  Start Accessory
                </p>
              }
            />
            <TableCell
              content={
                <ItemDisplay
                  isTruncated={false}
                  text="Display 1"
                />
              }
              data-terra-clinical-item-collection-content="display"
            />
            <TableCell
              content={
                <ItemDisplay
                  isTruncated={false}
                  text="Display 2"
                />
              }
              data-terra-clinical-item-collection-content="display"
            />
            <TableCell
              content={
                <ItemComment
                  isTruncated={false}
                  text="test comment"
                />
              }
              data-terra-clinical-item-collection-content="comment"
            />
            <TableCell
              className="content-accessory"
              content={
                <p>
                  End Accessory
                </p>
              }
            />
          </TableRow>
          <TableRow
            isSelectable={true}
            isSelected={false}
          >
            <TableCell
              className="content-accessory"
              content={
                <p>
                  Start Accessory
                </p>
              }
            />
            <TableCell
              content={
                <ItemDisplay
                  isTruncated={false}
                  text="Display 1"
                />
              }
              data-terra-clinical-item-collection-content="display"
            />
            <TableCell
              content={
                <ItemDisplay
                  isTruncated={false}
                  text="Display 2"
                />
              }
              data-terra-clinical-item-collection-content="display"
            />
            <TableCell
              content={
                <ItemComment
                  isTruncated={false}
                  text="test comment"
                />
              }
              data-terra-clinical-item-collection-content="comment"
            />
            <TableCell
              className="content-accessory"
              content={
                <p>
                  End Accessory
                </p>
              }
            />
          </TableRow>
        </SingleSelectableRows>
      </Table>
    </div>
  }
  responsiveTo="parent"
/>
`;

exports[`should render with multiple rows 1`] = `
<ResponsiveElement
  defaultElement={
    <div
      data-terra-clinical-item-collection-list-view={true}
    >
      <SingleSelectList
        hasChevrons={false}
        isDivided={false}
        onChange={[Function]}
      >
        <ListItem
          content={
            <ItemView
              accessoryAlignment="alignCenter"
              comment={
                <ItemComment
                  isTruncated={false}
                  text="test comment"
                />
              }
              displays={
                Array [
                  <ItemDisplay
                    isTruncated={false}
                    text="Display 1"
                  />,
                  <ItemDisplay
                    isTruncated={false}
                    text="Display 2"
                  />,
                ]
              }
              endAccessory={
                <p>
                  End Accessory
                </p>
              }
              isTruncated={false}
              layout="oneColumn"
              startAccessory={
                <p>
                  Start Accessory
                </p>
              }
              textEmphasis="default"
            />
          }
          isSelected={false}
        />
        <ListItem
          content={
            <ItemView
              accessoryAlignment="alignCenter"
              comment={
                <ItemComment
                  isTruncated={false}
                  text="test comment"
                />
              }
              displays={
                Array [
                  <ItemDisplay
                    isTruncated={false}
                    text="Display 1"
                  />,
                  <ItemDisplay
                    isTruncated={false}
                    text="Display 2"
                  />,
                ]
              }
              endAccessory={
                <p>
                  End Accessory
                </p>
              }
              isTruncated={false}
              layout="oneColumn"
              startAccessory={
                <p>
                  Start Accessory
                </p>
              }
              textEmphasis="default"
            />
          }
          isSelected={false}
        />
        <ListItem
          content={
            <ItemView
              accessoryAlignment="alignCenter"
              comment={
                <ItemComment
                  isTruncated={false}
                  text="test comment"
                />
              }
              displays={
                Array [
                  <ItemDisplay
                    isTruncated={false}
                    text="Display 1"
                  />,
                  <ItemDisplay
                    isTruncated={false}
                    text="Display 2"
                  />,
                ]
              }
              endAccessory={
                <p>
                  End Accessory
                </p>
              }
              isTruncated={false}
              layout="oneColumn"
              startAccessory={
                <p>
                  Start Accessory
                </p>
              }
              textEmphasis="default"
            />
          }
          isSelected={false}
        />
      </SingleSelectList>
    </div>
  }
  responsiveTo="parent"
  small={
    <div
      data-terra-clinical-item-collection-table-view={true}
    >
      <Table
        isPadded={true}
        isStriped={true}
        style={
          Object {
            "tableLayout": "fixed",
          }
        }
      >
        <TableHeader
          style={
            Object {
              "visibility": "hidden",
            }
          }
        >
          <TableHeaderCell
            columnType="accessory"
          />
          <TableHeaderCell
            columnType="display"
          />
          <TableHeaderCell
            columnType="display"
          />
          <TableHeaderCell
            columnType="comment"
          />
          <TableHeaderCell
            columnType="accessory"
          />
        </TableHeader>
        <SingleSelectableRows
          onChange={[Function]}
        >
          <TableRow
            isSelectable={true}
            isSelected={false}
          >
            <TableCell
              className="content-accessory"
              content={
                <p>
                  Start Accessory
                </p>
              }
            />
            <TableCell
              content={
                <ItemDisplay
                  isTruncated={false}
                  text="Display 1"
                />
              }
              data-terra-clinical-item-collection-content="display"
            />
            <TableCell
              content={
                <ItemDisplay
                  isTruncated={false}
                  text="Display 2"
                />
              }
              data-terra-clinical-item-collection-content="display"
            />
            <TableCell
              content={
                <ItemComment
                  isTruncated={false}
                  text="test comment"
                />
              }
              data-terra-clinical-item-collection-content="comment"
            />
            <TableCell
              className="content-accessory"
              content={
                <p>
                  End Accessory
                </p>
              }
            />
          </TableRow>
          <TableRow
            isSelectable={true}
            isSelected={false}
          >
            <TableCell
              className="content-accessory"
              content={
                <p>
                  Start Accessory
                </p>
              }
            />
            <TableCell
              content={
                <ItemDisplay
                  isTruncated={false}
                  text="Display 1"
                />
              }
              data-terra-clinical-item-collection-content="display"
            />
            <TableCell
              content={
                <ItemDisplay
                  isTruncated={false}
                  text="Display 2"
                />
              }
              data-terra-clinical-item-collection-content="display"
            />
            <TableCell
              content={
                <ItemComment
                  isTruncated={false}
                  text="test comment"
                />
              }
              data-terra-clinical-item-collection-content="comment"
            />
            <TableCell
              className="content-accessory"
              content={
                <p>
                  End Accessory
                </p>
              }
            />
          </TableRow>
          <TableRow
            isSelectable={true}
            isSelected={false}
          >
            <TableCell
              className="content-accessory"
              content={
                <p>
                  Start Accessory
                </p>
              }
            />
            <TableCell
              content={
                <ItemDisplay
                  isTruncated={false}
                  text="Display 1"
                />
              }
              data-terra-clinical-item-collection-content="display"
            />
            <TableCell
              content={
                <ItemDisplay
                  isTruncated={false}
                  text="Display 2"
                />
              }
              data-terra-clinical-item-collection-content="display"
            />
            <TableCell
              content={
                <ItemComment
                  isTruncated={false}
                  text="test comment"
                />
              }
              data-terra-clinical-item-collection-content="comment"
            />
            <TableCell
              className="content-accessory"
              content={
                <p>
                  End Accessory
                </p>
              }
            />
          </TableRow>
        </SingleSelectableRows>
      </Table>
    </div>
  }
/>
`;

exports[`should render with one row 1`] = `
<ResponsiveElement
  defaultElement={
    <div
      data-terra-clinical-item-collection-list-view={true}
    >
      <SingleSelectList
        hasChevrons={false}
        isDivided={false}
        onChange={[Function]}
      >
        <ListItem
          content={
            <ItemView
              accessoryAlignment="alignCenter"
              comment={
                <ItemComment
                  isTruncated={false}
                  text="test comment"
                />
              }
              displays={
                Array [
                  <ItemDisplay
                    isTruncated={false}
                    text="Display 1"
                  />,
                  <ItemDisplay
                    isTruncated={false}
                    text="Display 2"
                  />,
                ]
              }
              endAccessory={
                <p>
                  End Accessory
                </p>
              }
              isTruncated={false}
              layout="oneColumn"
              startAccessory={
                <p>
                  Start Accessory
                </p>
              }
              textEmphasis="default"
            />
          }
          isSelected={false}
        />
      </SingleSelectList>
    </div>
  }
  responsiveTo="parent"
  small={
    <div
      data-terra-clinical-item-collection-table-view={true}
    >
      <Table
        isPadded={true}
        isStriped={true}
        style={
          Object {
            "tableLayout": "fixed",
          }
        }
      >
        <TableHeader
          style={
            Object {
              "visibility": "hidden",
            }
          }
        >
          <TableHeaderCell
            columnType="accessory"
          />
          <TableHeaderCell
            columnType="display"
          />
          <TableHeaderCell
            columnType="display"
          />
          <TableHeaderCell
            columnType="comment"
          />
          <TableHeaderCell
            columnType="accessory"
          />
        </TableHeader>
        <SingleSelectableRows
          onChange={[Function]}
        >
          <TableRow
            isSelectable={true}
            isSelected={false}
          >
            <TableCell
              className="content-accessory"
              content={
                <p>
                  Start Accessory
                </p>
              }
            />
            <TableCell
              content={
                <ItemDisplay
                  isTruncated={false}
                  text="Display 1"
                />
              }
              data-terra-clinical-item-collection-content="display"
            />
            <TableCell
              content={
                <ItemDisplay
                  isTruncated={false}
                  text="Display 2"
                />
              }
              data-terra-clinical-item-collection-content="display"
            />
            <TableCell
              content={
                <ItemComment
                  isTruncated={false}
                  text="test comment"
                />
              }
              data-terra-clinical-item-collection-content="comment"
            />
            <TableCell
              className="content-accessory"
              content={
                <p>
                  End Accessory
                </p>
              }
            />
          </TableRow>
        </SingleSelectableRows>
      </Table>
    </div>
  }
/>
`;
